RecoParticleFlow/PFClusterProducer/interface/PFRecHitQTests.h

```diff
@@ -381,6 +381,80 @@ class PFRecHitQTestHOThreshold : public PFRecHitQTestBase {
 };
 
 //
+//  Quality test that checks threshold as a function of ECAL eta-ring
+//
+#include "Calibration/Tools/interface/EcalRingCalibrationTools.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+class PFRecHitQTestECALThreshold : public PFRecHitQTestBase {
+ public:
+  PFRecHitQTestECALThreshold() {
+
+  }
+
+  PFRecHitQTestECALThreshold(const edm::ParameterSet& iConfig):
+    PFRecHitQTestBase(iConfig)
+    {
+      thresholds_ = iConfig.getParameter<std::vector<double> >("thresholds");
+    }
+
+    void beginEvent(const edm::Event& event,const edm::EventSetup& iSetup) {
+      edm::ESHandle<CaloGeometry> pG;
+      iSetup.get<CaloGeometryRecord>().get(pG);
+      CaloSubdetectorGeometry const* endcapGeometry = pG->getSubdetectorGeometry(DetId::Ecal, EcalEndcap);
+      endcapGeometrySet_=false;
+      if (endcapGeometry) {
+        EcalRingCalibrationTools::setCaloGeometry(&(*pG));
+        endcapGeometrySet_=true;
+      }
+    }
+
+    bool test(reco::PFRecHit& hit,const EcalRecHit& rh,bool& clean,bool fullReadOut){
+      return fullReadOut || pass(hit);
+    }
+    bool test(reco::PFRecHit& hit,const HBHERecHit& rh,bool& clean){
+      return true;
+    }
+
+    bool test(reco::PFRecHit& hit,const HFRecHit& rh,bool& clean){
+      return true;
+    }
+    bool test(reco::PFRecHit& hit,const HORecHit& rh,bool& clean){
+      return true;
+    }
+
+    bool test(reco::PFRecHit& hit,const CaloTower& rh,bool& clean){
+      return true;
+    }
+
+    bool test(reco::PFRecHit& hit,const HGCRecHit& rh,bool& clean){
+      return true;
+    }
+
+ protected:
+    std::vector<double> thresholds_;
+    bool endcapGeometrySet_;
+
+    bool pass(const reco::PFRecHit& hit){
+      
+      // this is to skip endcap ZS for Phase2 until there is a defined geometry
+      // apply the loosest ZS threshold, for the first eta-ring in EE
+      if(!endcapGeometrySet_) {
+        int firstEERing = 171;
```

Comment:
can this magic number be obtained from conditions or other configuration?
If not, please add some in-line comment clarifying how 171 is obtained.